### ISSUE TITLE
Incorrect completion when using nvim_input()/feedkeys()

### ISSUE THREAD

**Issue by lePerdu** [neovim/neovim#8480]
- `nvim --version`
NVIM v0.3.0-1295-g5b2cee03c
- Vim (version: 8.1) behaves differently?
no
- Operating system/version:
Void linux
- Terminal name/version:
termite
- `$TERM`:
xterm-termite

### Steps to reproduce using `nvim -u NORC`

With file contents
```
abc abdef
| <- cursor here
```

Typing `ab<C-N><BS>d` in insert mode gives 1 completion item: "abdef".
Calling `nvim_input('iab<C-N><BS>d')` (or an equivalent `feedkeys()` call) gives 2 completion items: "abd" and "abdef".

### Expected behaviour
I would guess that the word being completed should not be in the completion list, but in any case `nvim_input()` and actually typing the keys should behave the same.

**Comment by justinmk** [comment]
Yes, `nvim_input()` should work like real user input. Using tmux send-keys works:

    send-keys iab C-N Bspace d

For testing purposes, a workaround is to use `helpers.wait()` (which waits for input to "flush"):

```lua
nvim_input('iab<C-N>')
wait()
nvim_input('<BS>d')
```

**Comment by nwounkn** [comment]
fixed c9963e2212bda7a4d2186469ad07078d348353cf
